engine/engine.go

```diff
@@ -186,9 +186,11 @@ func (e *EngineAPI) ForkchoiceUpdatedV3(
 	//   It must return STATUS_VALID if all of the transactions could be executed without error.
 	// TODO checktx doesn't actually run the tx, it only does basic validation.
 	for _, txBytes := range cosmosTxs {
-		if _, err := e.txValidator.CheckTx(context.Background(), &abci.RequestCheckTx{
+		if checkTxResult, err := e.txValidator.CheckTx(context.Background(), &abci.RequestCheckTx{
 			Tx: txBytes,
-		}); err != nil {
+		}); err != nil { // a catch-all for errors unrelated to the actual validation.
+			return nil, engine.GenericServerError.With(fmt.Errorf("convert cosmos txs to eth txs: %v", err))
```

Comment:
I'm a little confused by the error message wrapper and feel like I may be missing something, where are we converting the cosmos txs to eth txs?

Reply:
Ah my bad. Copy-pasta mistake. Good catch!